bom.py

```diff
@@ -179,6 +162,33 @@ def __setup__(cls):
         for fname in ('effective_date', 'planned_date'):
             if fname not in cls.bom.depends:
                 cls.bom.depends.add(fname)
+        cls.bom.domain += [If((Eval('state').in_(['request', 'draft'])) & ~Bool(Eval('bom', False)),
+            [
+                ('start_date', '<=', If(Bool(Eval('effective_date')), Eval('effective_date'), Eval('planned_date', Date()))),
+                ['OR',
+                    ('end_date', '>=', If(Bool(Eval('effective_date')), Eval('effective_date'), Eval('planned_date', Date()))),
+                    ('end_date', '=', None),
+                ]
+            ],
+            ())]
+
+    @classmethod
+    def run(cls, productions):
+        pool = Pool()
+        Warning = pool.get('res.user.warning')
+
+        for production in productions:
+            if production.bom and not production.bom.is_valid:
+                production_date = production.effective_date or production.planned_date
+                with Transaction().set_context(production_date=production_date):
+                    key = 'bom_expired_date_%s_%s' % (production.id, production.bom.id)
```

Comment:
Use dates instead of "is valid" and remove the "is_valid" property. Also remove production_date from domain in __setup__

Reply:
See 9daf0f4ff75896ec485cc9785c5fe47e8520b9db